skbio/io/format/fastq.py

```diff
@@ -317,7 +317,11 @@ def _fastq_sniffer(fh):
 
     try:
         not_empty = False
-        for _ in zip(range(10), _fastq_to_generator(fh, phred_offset=33)):
+        for _, seq in zip(range(10), _fastq_to_generator(fh, phred_offset=33)):
+            if len((seq.metadata['id'] +
+                    seq.metadata['description']).split(':')) == 10 and \
+               seq.metadata['description'].split(':')[1] in 'YN':
```

Comment:
This code is a little messy (the indentation especially, though you have satisfied PEP8!). It might be worth putting some temp variables in instead of making it one long expression.